ipamd/ipamd.go

```diff
@@ -628,15 +631,17 @@ func (c *IPAMContext) tryAllocateENI() {
 	if err != nil {
 		ipamdErrInc("increaseIPPoolwaitENIAttachedFailed")
 		log.Errorf("Failed to increase pool size: Unable to discover attached ENI from metadata service %v", err)
-		return
+		return err
 	}
 
 	err = c.setupENI(eni, eniMetadata)
 	if err != nil {
 		ipamdErrInc("increaseIPPoolsetupENIFailed")
 		log.Errorf("Failed to increase pool size: %v", err)
-		return
+		return err
```

Comment:
Maybe you can simply return the error in the above line instead of logging it here and let the caller take care of logging that error?